src/dfx/src/commands/ledger/create_canister.rs

```diff
@@ -74,13 +78,21 @@ pub async fn exec(env: &dyn Environment, opts: CreateCanisterOpts) -> DfxResult
         )
     })?;
 
+    let created_at_time = opts
+        .created_at_time
+        .as_ref()
+        .map_or(Ok(None), |v| {
+            u64::from_str_radix(v, 10).map(|n| Some(n)).map_err(|err| anyhow!(err))
+        })
+        .context("Failed to parse nanoseconds created_at_time.")?;
+
```

Comment:
If `Option<String>` is required, please DRY this (same code block exists in three places)

Reply:
String is not required. Changed to Option<u64> and removed the conversions.